src/components/modal/nz-modal-subject.service.ts

```diff
@@ -13,7 +13,7 @@ const enum modalEvent {
 }
 
 @Injectable()
-export class NzModalSubject extends Subject<string> {
+export class NzModalSubject extends Subject<string | object> {
```

Comment:
Better to use `any` if indeed accept any type from user, didn't know it was a user defined type. This way it still cannot accept `boolean` & `number` type.